### PR TITLE
Update header and query parameter for source control

### PR DESCRIPTION
**Pull request by Juliehzl**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [x] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories")
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json [hunk 1]
```diff
@@ -230,6 +237,27 @@
...
+            "required": false,
+            "type": "boolean"
+          },
+          {
+            "name": "deleteWorkflow",
+            "in": "query",
+            "description": "Delete workflow.",
+            "required": false,
+            "type": "boolean"
+          },
```
Comment (ramoka178) [inline specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json:260]: Please explain on what are these ?

If the API is deleting sourcecontrol resource, why are these needed ? 
Can it be a different API to delete workflows ?

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json [hunk 2]
```diff
@@ -171,6 +171,13 @@
               "$ref": "#/definitions/SourceControl"
             }
           },
+          {
+            "name": "x-ms-github-auxiliary",
```
Comment (ramoka178) [inline specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json:175]: >x-ms-github-auxiliary [](http://example.com/codeflow?start=21&length=21)

who sends this header ? and curious on what/how is it used ?

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json [hunk 3]
```diff
@@ -171,6 +171,13 @@
...
+            "in": "header",
+            "description": "Github personal access token used for SourceControl.",
+            "required": true,
+            "type": "string"
+          },
```
Comment (ramoka178) [inline specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json:180]: Is this API version on this PUT API already open to public and code is receiving this header ? 
And this is just to update the swagger ?

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json [hunk 4]
```diff
@@ -171,6 +171,13 @@
...
           {
             "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
```
Comment (ramoka178) [inline specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json:182]: >v3 [](http://example.com/codeflow?start=69&length=2)

update to v5 here and at all places in this file for common-types

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json (hunk 1 above)
Comment (ramoka178) [inline specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json:260]: Update the examples to reflect these

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json (hunk 2 above)
Comment (Juliehzl) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json:175]: customer needs to provide the header to pass in the Github token for API backend implementation requirement

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json (hunk 3 above)
Comment (Juliehzl) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json:180]: In backend code, from day one we will require the header. Here is to fix the issue in the swasgger of new api version.

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json (hunk 1 above)
Comment (Juliehzl) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json:260]: When creating source control resource, we will create a github workflow for customer. When deleting source control, we give customer the option to only delete source control resource or delete source control and created workflow together.

There is no need to delete workflows because it is a backend resource and not exposed to customer directly, which is also the design from day one.

Comment (ramoka178) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json:260]: Update examples to reflect ignoreWorkflowDeletionFailure and deleteWorkflow.

Comment (ramoka178) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json:260]: So, are these working APIs today that the customers are using  with ignoreWorkflowDeletionFailure and deleteWorkflow query params ? 

Delete is supposed to delete only a single resource.

Could you post any approval that these query parameters could be used ?

Comment (harryli0108) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json:260]: In this case, we're deleting single resource (which is the SourceControl proxy resource). The option to delete the GH workflow file is just a consolidated option for convenience for the user to cleanup any leftover files.

Comment (Juliehzl) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json:260]: updated

Comment (ramoka178) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json:260]: [ARMBlockingComment] If GH work flow is implicitly created on a source control creation, the deletion of source control resource should also result in deleting GH workflow . There should not be an extra query param for user to pass to clean up GH workflow. 

Please schedule a slot in ARM API Review office hours that happens every day to discuss further on this.
https://eng.ms/docs/products/arm/introduction/office_hours

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json (hunk 4 above)
Comment (razvanbadea-msft) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json]: For consistency you should also update the rest of the parameters to v5

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json (hunk 1 above)
Comment (razvanbadea-msft) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json:260]: Did you get a review for this matter?

Comment (Juliehzl) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json:260]: yes

**Comment by Juliehzl** [comment]
same change with https://github.com/Azure/azure-rest-api-specs/pull/29570. please signoff

**Review by ms-bogdan-rotaru — APPROVED** [review]
